### ISSUE TITLE
Clean code, spectral_norm, and GAN losses?

### ISSUE THREAD

**Issue by Johnson-yue** [hologerry/AGIS-Net#4]
Hi,  your work is very interesting and I will to learn it, than I read your code it is huge number.

Would you remove some unnecessary code ? like in `networks.py` I think you only use `"agisnet"` and not Use **any U-net** so,  maybe remove them is better for reading. like : 

networks.py :Line 117-141 and some

And Second Question : why do you not use spectral_norm  for stability training??

**Comment by hologerry** [comment]
First question: We think the U-Net code is helpful for someone who want to learn the difference between our model and U-Net, and the U-Net code is a good example to write some new U-Net like models.

Second question: Good question! Actually, we tried, but it did not work as expected with the default hyper-parameters in original paper, so we abandoned it. You can try to do more hyper-parametes tuning to make it work.

**Comment by Johnson-yue** [comment]
ok, and other Loss Function did you try like Hinge Loss, LS loss, Ra_LS loss ,and so on ? the standard GAN loss was better performance than others?

**Comment by hologerry** [comment]
Nope, you can also try others.
For a (semi-)supervised task, we focused on the model architecture and task-specific loss. The performance on different general GAN losses might be different for a specific task. However, it will take too much time to choose the 'right' loss, so we just use the standard one, in the end, its effect is not bad.

**Comment by Johnson-yue** [comment]
ok ,thank your reply
